### ISSUE TITLE
[Stretch] Network attack simulation

### ISSUE THREAD

**Issue by robzajac** [libp2p/py-libp2p#57]
The Ethereum networking layer has been attacked several times. Py-libp2p and more generally libp2p would benefit from protection against [common attacks on peer to peer networks](http://www.dcs.gla.ac.uk/~sadekf/P2PSecTaxonomy.pdf). Py-libp2p should simulate attacks on topologies constructed through this library and assess security against these attacks.

**Comment by zixuanzh** [comment]
We can start from eclipse attack simulation as described in this [paper] (https://eprint.iacr.org/2018/236.pdf). It is done through poisoning the DHT which is the primary source of structure in the network.

**Comment by mgoelzer** [comment]
@lanzafame Yes, I agree.  We are building a test cluster project this quarter (reminder to self:  insert issue link here once spec issue is available), then we will definite interop tests for common modules like DHT, PubSub, etc.  I made https://github.com/libp2p/interop/issues/3 to track the need to bring github.com/libp2p/interop up to date or move that content into another repo.

**Comment by acul71** [comment]
Closing as obsolete/completed for tracker purposes. Network attack simulation coverage exists in-tree under `tests/examples/attack_simulation/`.
